### PR TITLE
Added BARCODE_TAG option.

### PR DESCRIPTION
**Pull request by jhl667**
Included help section description and optional BARCODE_TAG option available in 2.7.1.

### CODE REVIEW

File: tools/picard/picard_MarkDuplicates.xml [hunk 1]
```diff
@@ -53,6 +59,8 @@
     </param>
     <param name="optical_duplicate_pixel_distance" type="integer" value="100" min="0" max="500" label="The maximum offset between two duplicte clusters in order to consider them optical duplicates" help="OPTICAL_DUPLICATE_PIXEL_DISTANCE; default=100"/>
 
+    <param name="barcode_tag" type="text" optional="True" default="null" label="Barcode Tag" help="Barcode SAM tag. This tag can be utilized when you have data from an assay that includes Unique Molecular Indices."/>
```
Comment (bgruening) [inline tools/picard/picard_MarkDuplicates.xml]: Can you remove the default attribute, this is not needed.

File: tools/picard/picard_MarkDuplicates.xml [hunk 2]
```diff
@@ -26,6 +27,11 @@
     READ_NAME_REGEX=${ pipes.quote( str( $read_name_regex ) ) or "''" }
     OPTICAL_DUPLICATE_PIXEL_DISTANCE="${optical_duplicate_pixel_distance}"
 
+    # Optional arguments
+    #if $barcode_tag:
+    BARCODE_TAG="${barcode_tag}"
```
Comment (bgruening) [inline tools/picard/picard_MarkDuplicates.xml]: Please use single quotes.

Comment (jhl667) [inline tools/picard/picard_MarkDuplicates.xml]: Would you like me to change some of the other parameters in this tool from double to single quotes?

Comment (bgruening) [inline tools/picard/picard_MarkDuplicates.xml]: This would be super cool! If you find a citation as well, you can also add this as macro.
Thanks!

Comment (jhl667) [inline tools/picard/picard_MarkDuplicates.xml]: I think a citation macro would be great, but Picard still doesn't have a paper.  Is there a way to cite a webpage using the citation tag?

**Comment by jhl667** [comment]
@bgruening I believe the suggested modifications have been taken care of?  Anything else you need from this PR?

**Comment by bgruening** [comment]
@jhl667 the tests do not pass because of linting error. Could you fix this as well?
https://travis-ci.org/galaxyproject/tools-iuc/jobs/215662653#L495

**Comment by jhl667** [comment]
@bgruening Sorry about the delay, I've located a suitable replacement for the broken links.

**Comment by bgruening** [comment]
@jhl667 cool thanks!

**Pull request merged**